packages/inertia/src/inertia.js

```diff
@@ -170,8 +226,12 @@ export default {
           return this.setPage(error.response.data, { visitId })
         } else if (Axios.isCancel(error)) {
           onCancel()
-        } else if (this.isHardVisit(error.response)) {
-          this.hardVisit(error.response.headers['x-inertia-location'])
+        } else if (this.isLocationVisitResponse(error.response)) {
+          let locationUrl = this.normalizeUrl(error.response.headers['x-inertia-location'])
+          if (url.hash && this.urlWithoutHash(url) === this.urlWithoutHash(locationUrl)) {
+            locationUrl.hash = url.hash
+          }
+          this.locationVisit(locationUrl.href, preserveScroll)
```

Comment:
Is this even necessary? If we're not passing in the hash, I don't think we need to check it either?